rack-protection/lib/rack/protection/authenticity_token.rb

```diff
@@ -12,11 +12,74 @@ module Protection
     # Only accepts unsafe HTTP requests if a given access token matches the token
     # included in the session.
     #
-    # Compatible with rack-csrf.
+    # Compatible with the {rack-csrf}[https://rubygems.org/gems/rack_csrf] gem.
     #
-    # Options:
+    # The middleware only accepts unsafe HTTP requests (exclude <tt>GET</tt>,
+    # <tt>HEAD</tt>, <tt>OPTIONS</tt>, <tt>TRACE</tt>) if a given access token
+    # matches the token included in the session.
+    # 
+    # It checks either <tt>X-CSRF-Token</tt> header or <tt>POST</tt> form data.
     #
-    # authenticity_param: Defines the param's name that should contain the token on a request.
+    # == Options
+    #
+    # [authenticity_param] Defines the name of the param that should contain
+    #                      the token on a request.
+    #
+    # == Example: Using option <tt>authenticity_token</tt>
+    #
+    # To customize the authenticity parameter for form data (default is <tt>"authenticity_token"</tt>):
+    #   use Rack::Protection::AuthenticityToken, authenticity_param: 'your_param_name'
+    #
+    # == Example: Forms application
+    #
+    # To show what the AuthenticityToken does, this section includes a sample
+    # program which shows two forms. One with, and one without a CSRF token
+    # The one without CSRF token field will get a 403 Forbidden response.
+    #
+    # Install the gem, then run the program:
+    #
+    #   gem install 'rack-protection'
+    #   ruby server.rb
+    #
+    # Here is <tt>server.rb</tt>:
+    #
+    #  require 'rack/protection'
+    #  csrf_param = 'token'
+    #
+    #  app = Rack::Builder.app do
+    #    use Rack::Session::Cookie, secret: 'secret'
+    #    use Rack::Protection::AuthenticityToken, authenticity_param: csrf_param
```

Comment:
For a basic example, I think I would leave off the `authenticity_param` option and use the default value.